### PR TITLE
Generate RunData from Laravel and pass to Blade

### PR DESCRIPTION
**Pull request by danieltharp**
This PR creates a new `generateScreenVars()` method in Legacy Tools that runs through the entire process of bootstrapping a RunData instance and compiling all the needed information to render a page, and pass it to a Blade template as a set of Laravel-compatible variables. It also has a sample route using this method as a proof of concept.

There is a fair amount of fat that could be trimmed but I wanted to merge what I've got so far into develop as it's a self-contained addition, only affecting the "What is a Wiki?" page from the default sidebar.

### CODE REVIEW

File: web/app/Helpers/LegacyTools.php [hunk 1]
```diff
@@ -41,4 +60,520 @@ public static function isSystemAccount(int $id) : bool
...
+        $c->add("site.deleted", false);
+        $site = SitePeer::instance()->selectOne($c);
+        if ($site === null) {
+            $c = new Criteria();
+            $c->add("custom_domain", $siteHost);
+            $c->add("site.deleted", false);
+
+            /** @var Site? $site */
+            $site = SitePeer::instance()->selectOne($c);
+        }
```
Comment (emmiegit) [inline web/app/Helpers/LegacyTools.php:97]: This domain check functionality will be useful elsewhere, at some point I would suggest pulling it out into its own utility.

File: web/app/Helpers/LegacyTools.php [hunk 2]
```diff
@@ -41,4 +60,520 @@ public static function isSystemAccount(int $id) : bool
...
+        if ($site === null) {
+            // check for redirects
+            $c = new Criteria();
+            $q = "SELECT site.* FROM site, domain_redirect WHERE domain_redirect.url='".db_escape_string($siteHost)."' " .
+                "AND site.deleted = false AND site.site_id = domain_redirect.site_id LIMIT 1";
+            $c->setExplicitQuery($q);
+            /** @var Site? $site */
+            $site = SitePeer::instance()->selectOne($c);
+            if ($site !== null) {
+                $newUrl = GlobalProperties::$HTTP_SCHEMA . "://" . $site->getDomain().$_SERVER['REQUEST_URI'];
```
Comment (emmiegit) [inline web/app/Helpers/LegacyTools.php:107]:
```suggestion
                $newUrl = GlobalProperties::$HTTP_SCHEMA . "://" . $site->getDomain() . $_SERVER['REQUEST_URI'];
```

File: web/app/Helpers/LegacyTools.php [hunk 3]
```diff
@@ -41,4 +60,520 @@ public static function isSystemAccount(int $id) : bool
...
+                header("HTTP/1.1 301 Moved Permanently");
+                header("Location: ".$newUrl);
+                exit();
+            }
+        }
+
+        if ($site === null) {
+            // echo file_get_contents(WIKIJUMP_ROOT."/resources/views/site_not_exists.html");
```
Comment (emmiegit) [inline web/app/Helpers/LegacyTools.php:115]: Commented code

File: web/app/Helpers/LegacyTools.php [hunk 4]
```diff
@@ -41,4 +60,520 @@ public static function isSystemAccount(int $id) : bool
...
+            }
+        }
+
+        /**
+         * Process Modules
+         */
+        $runData->setTemp("jsInclude", array());
+        // process modules...
+        $moduleProcessor = new ModuleProcessor($runData);
+        //$moduleProcessor->setJavascriptInline(true); // embed associated javascript files in <script> tags
```
Comment (emmiegit) [inline web/app/Helpers/LegacyTools.php:329]: Why is this commented out?

File: web/app/Helpers/LegacyTools.php [hunk 5]
```diff
@@ -41,4 +60,520 @@ public static function isSystemAccount(int $id) : bool
...
+            // ?
+            // scripts
+        $rendered = preg_replace(';<script(.*?)src="'.GlobalProperties::$HTTP_SCHEMA . "://" . GlobalProperties::$URL_HOST_PREG . '(.*?)</script>;s', '<script\\1src="https://' . GlobalProperties::$URL_HOST . '\\2</script>', $rendered);
+        $rendered = preg_replace(';<link(.*?)href="'.GlobalProperties::$HTTP_SCHEMA . "://" . GlobalProperties::$URL_HOST_PREG . '(.*?)/>;s', '<link\\1href="https://' . GlobalProperties::$URL_HOST . '\\2/>', $rendered);
+        $rendered = preg_replace(';(<img\s+.*?src=")http(://' . GlobalProperties::$URL_HOST_PREG . '(.*?)/>);s', '\\1https\\2', $rendered);
+        do {
+        $renderedOld = $rendered;
+        $rendered = preg_replace(';(<style\s+[^>]*>.*?@import url\()http(://' . GlobalProperties::$URL_HOST_PREG . '.*?</style>);si', '\\1https\\2', $rendered);
+        } while ($renderedOld != $rendered);
+        }
```
Comment (emmiegit) [inline web/app/Helpers/LegacyTools.php:357]: This section is indented very weirdly.

File: web/app/Helpers/LegacyTools.php [hunk 6]
```diff
@@ -41,4 +60,520 @@ public static function isSystemAccount(int $id) : bool
...
+
+        echo str_replace("%%%CURRENT_TIMESTAMP%%%", (string)time(), $rendered);
+
+//        dd($rendered);
```
Comment (emmiegit) [inline web/app/Helpers/LegacyTools.php:361]: What's this?

File: web/app/Helpers/LegacyTools.php [hunk 7]
```diff
@@ -41,4 +60,520 @@ public static function isSystemAccount(int $id) : bool
...
+            $lang = $user->language;
+
+            switch ($lang) {
+                case 'pl':
+                    $glang="pl_PL";
+                    break;
+                case 'en':
+                    $glang="en_US";
+                    break;
+            }
```
Comment (emmiegit) [inline web/app/Helpers/LegacyTools.php:432]: I swear this switch appears like 20 times in wikidot's code lol.

File: web/app/Helpers/LegacyTools.php [hunk 8]
```diff
@@ -41,4 +60,520 @@ public static function isSystemAccount(int $id) : bool
...
+
+        $return['login'] = $out;
+
+        /**
+         * Page Options Bottom module injection
+         */
+        $pl = $runData->getParameterList();
+        $pageName = $runData->getTemp("pageUnixName");
+
+        $page = $runData->getTemp("page");//$pl->getParameterValue("page", "MODULE");
```
Comment (emmiegit) [inline web/app/Helpers/LegacyTools.php:486]: What's this commented out bit?

**Comment by danieltharp** [comment]
One thing here, most of the code in the new LegacyTools method was brought straight over from the various RunData methods that comprise it. The cleanup on that is to come in another PR.

**Review by emmiegit — APPROVED** [review]

**Comment by emmiegit** [comment]
Thanks for all the work on this!

**Comment by emmiegit** [comment]
> One thing here, most of the code in the new LegacyTools method was brought straight over from the various RunData methods that comprise it. The cleanup on that is to come in another PR.

You can ignore such comments from my review then.

**Pull request merged**